### ISSUE TITLE
TCNForecaster should add validation set option for fit method

### ISSUE THREAD

**Issue by TheaperDeng** [intel/analytics-zoo#333]
Now we use training data as the validation set.
